packages/arb-bridge-eth/contracts/arch/OneStepProof.sol

```diff
@@ -20,922 +20,772 @@ pragma solidity ^0.5.11;
-        Value.Data memory val2
-    ) internal pure returns (bool) {
-        machine.addDataStackValue(Value.newBoolean(val1.hash() == val2.hash()));
-        return true;
+    function executeEqInsn(AssertionContext memory context) internal pure {
+        Value.Data memory val1 = context.stack.popVal();
+        Value.Data memory val2 = context.stack.popVal();
+        context.stack.pushVal(Value.newBoolean(val1.hash() == val2.hash()));
     }
 
-    function executeIszeroInsn(
-        Machine.Data memory machine,
-        Value.Data memory val1
-    ) internal pure returns (bool) {
+    function executeIszeroInsn(AssertionContext memory context) internal pure {
+        Value.Data memory val1 = context.stack.popVal();
         if (!val1.isInt()) {
-            machine.addDataStackInt(0);
+            context.stack.pushVal(Value.newInt(0));
         } else {
             uint256 a = val1.intVal;
             uint256 c;
             assembly {
                 c := iszero(a)
             }
-            machine.addDataStackInt(c);
+            context.stack.pushVal(Value.newInt(c));
         }
-        return true;
     }
 
-    function executeAndInsn(
-        Machine.Data memory machine,
-        Value.Data memory val1,
-        Value.Data memory val2
-    ) internal pure returns (bool) {
-        if (!val1.isInt() || !val2.isInt()) {
-            return false;
+    function executeNotInsn(AssertionContext memory context) internal pure {
+        Value.Data memory val1 = context.stack.popVal();
+        if (!val1.isInt()) {
+            context.handleOpcodeError();
+            return;
         }
         uint256 a = val1.intVal;
-        uint256 b = val2.intVal;
         uint256 c;
         assembly {
-            c := and(a, b)
+            c := not(a)
         }
-        machine.addDataStackInt(c);
-        return true;
+        context.stack.pushVal(Value.newInt(c));
     }
 
-    function executeOrInsn(
-        Machine.Data memory machine,
-        Value.Data memory val1,
-        Value.Data memory val2
-    ) internal pure returns (bool) {
-        if (!val1.isInt() || !val2.isInt()) {
-            return false;
-        }
-        uint256 a = val1.intVal;
-        uint256 b = val2.intVal;
-        uint256 c;
-        assembly {
-            c := or(a, b)
-        }
-        machine.addDataStackInt(c);
-        return true;
-    }
+    /* solhint-enable no-inline-assembly */
 
-    function executeXorInsn(
-        Machine.Data memory machine,
-        Value.Data memory val1,
-        Value.Data memory val2
-    ) internal pure returns (bool) {
-        if (!val1.isInt() || !val2.isInt()) {
-            return false;
-        }
-        uint256 a = val1.intVal;
-        uint256 b = val2.intVal;
-        uint256 c;
-        assembly {
-            c := xor(a, b)
-        }
-        machine.addDataStackInt(c);
-        return true;
+    // Hash
+
+    function executeSha3Insn(AssertionContext memory context) internal pure {
```

Comment:
Suggest renaming this, because the instruction isn't called sha3, and it doesn't compute the official sha3 algorithm.

Reply:
Renamed to executeHashInsn